python/heterocl/compute_api.py

```diff
@@ -921,5 +924,67 @@ def reduce_body():
     make_reduce.__doc__ = doc_str.format(name)
     return make_reduce
 
+
+def bitcast(tensor, dst_dtype):
+    """Bitcast a HeteroCL tensor to the destination data type of the same bitwidth.
+
+    This API **bitcast** the input tensor from its own data type (source dtype)
+    to the destination data type (dst_dtype). The destination data type must have
+    the same bitwidth with the source datatype. 
+
+    Parameters
+    ----------
+    tensor: Tensor
+        The input tensor of the source data type
+    dst_dtype: Type
+        The destination data type. For example, hcl.UInt(32)
+
+    Returns
+    -------
+    _tensor: Tensor
+        The bitcasted tensor of destination data ype
+    """
+    
+    # check type
+    if not isinstance(tensor, Tensor):
+        raise APIError("bitcast only support HeteroCL Tensor")
+
+    if not isinstance(dst_dtype, Type):
+        raise APIError("dst_dtype should be HeteroCL data type")
+
+    # check bitwidth
+    src_bitwidth = util.get_type(tensor.dtype)[1]
+    dst_bitwidth = dst_dtype.bits 
+    if src_bitwidth != dst_bitwidth:
+        raise APIError("[compute_api.bitcast] Destination bitwidth is not equal to source bitwidth. " + 
+            "The destination data type must have the same bitwidth with the source datatype.") 
+
+    shape = tensor.shape
+    name = get_name("bitcast", tensor.name)
+    dst_dtype_str = dtype_to_str(dst_dtype)
+
+    def _iter_tensor(_tensor, tensor, buffer_var):
+        for indices in list(product(*[range(x) for x in tensor.shape])):
+            index, _, _ = get_index(shape, indices, 0)
+            load = _make.Load(tensor.dtype, tensor.buf.data, index)
+            expr = _make.Call(dst_dtype_str, "bitcast", [tensor[indices]], _expr.Call.PureIntrinsic, None, 0)
+            stage.emit(
+                _make.Store(
+                    buffer_var,
+                    _make.Cast(stage._dtype, expr),
+                    index
+                )
+            )
+
```

Comment:
You should create a `for` loop instead.  [Here](https://github.com/cornell-zhang/heterocl/blob/bbfdae4abbbe9268c65cdcad7b8d312412cf1248/python/heterocl/compute_api.py#L391) is an example of using `make_for`. Otherwise, you are basically unrolling the loop by default.